BrainlyDemoDay/BrainlyDemoDay/SubscriptionManager/SubscriptionManager.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+public enum SubscriptionState {
+    case monthly(Date)
+    case limitedEntries(Int)
+    case empty
+}
+
+public protocol SubscriptionManagerProtocol {
+    func fetchSubscriptionFor(tutoringUser user: TutoringSDKUser, completion: @escaping (Swift.Result<SubscriptionState, Error>) -> Void)
```

Comment:
I would imagine that this specific method returns some kind of `TutoringSubscription` or generic `Subscription` object, which can be interpreted by the logic. This object would have a state information and/or properties that are coupled with a specific type of subscription.